### PR TITLE
astar_avoid: handle error returned from calling libwaypointfollower::getClosestWaypoint()

### PR DESCRIPTION
**Pull request by icolwell-as**
Must be merged before #36 

For detailed explanation, see https://gitlab.com/astuff/autoware.ai/core_planning/-/issues/39

See original MR for more details: https://gitlab.com/astuff/autoware.ai/core_planning/-/merge_requests/71

### CODE REVIEW

File: waypoint_planner/src/astar_avoid/astar_avoid.cpp [hunk 1]
```diff
@@ -370,35 +374,34 @@ tf::Transform AstarAvoid::getTransform(const std::string& from, const std::strin
...
+int AstarAvoid::getLocalClosestWaypoint(const autoware_msgs::Lane& waypoints, const geometry_msgs::Pose& pose,
+                                        const int& search_size)
 {
-  static int prev_index = -1;
-  static autoware_msgs::Lane local_waypoints;  // around self-vehicle
-
   // search in all waypoints if lane_select judges you're not on waypoints
   if (closest_waypoint_index_ == -1)
   {
     prev_index = -1;
```
Comment (JWhitleyWork) [inline waypoint_planner/src/astar_avoid/astar_avoid.cpp:383]: Removing line 375 above causes this variable to be undefined when it is used here. This is causing CI failures. Please correct.

Comment (icolwell-as) [reply waypoint_planner/src/astar_avoid/astar_avoid.cpp]: Thanks for catching! fixed.

**Comment by icolwell-as** [comment]
@JWhitleyWork , I just found out about an issue with this PR and #36. See [here](https://gitlab.com/astuff/autoware.ai/core_planning/-/issues/68) for details if interested. 

Anyway, I'm going to pause work on these two PRs until I sort out a solution.

**Comment by icolwell-as** [comment]
@JWhitleyWork, this PR and #36 should now be fixed. 
I'll update with master once #23 is merged.

**Pull request merged**